File: src/main/java/com/hyak4j/student/Student.java
package com.hyak4j.student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest(){
//        int max = (english > math) ? english : math;
//        if (english > math){
//            max = english;
//        }else {
//            max = math;
//        }
        return (english > math) ? english : math;
    }

    public void print(){
        System.out.println(name + "\t" + english + "\t" + math +
                "\t" + getAverage() + "\t" +
                ((getAverage() >= 60) ? "PASS" : "FAILED"));
        char grading = 'F';
        switch (getAverage()/10){
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'F';
                break;
        }
/*        if (getAverage() >= 90 && getAverage() <= 100){
            grading = 'A';
        }else if (getAverage() >= 80 && getAverage() <= 89){
            grading = 'B';
        }else if (getAverage() >= 70 && getAverage() <= 79){
            grading = 'C';
        }else if (getAverage() >= 60 && getAverage() <= 69){
            grading = 'D';
        }*/

        System.out.println(grading);


/*        if (getAverage() >= 60){
            System.out.println("\tPASS");
        }else {
            System.out.println("\tFAILED");
        }*/
    }

    public int getAverage(){
        return (english + math)/2;
    }

}

File: src/main/kotlin/com/hyak4j/kotlin/student/Student.kt
package com.hyak4j.kotlin.student

import java.util.Scanner

fun main() {
//    userInput()
    Student.pass = 10
    val student = Student("Henry", 80, 90)
    val student2 = Student("Tom", 30, 20)
    val student3 = Student("James", 50, 40)

    val gstu = GraduateStudent("Eric", 60, 60, 60)
    gstu.print()
    student.print()
    student2.print()
    student3.print()
    println("High score: ${student.highest()}")
}

private fun userInput() {
    val scanner = Scanner(System.`in`)
//    println("Please enter student`s name: ")
    var name = null
    println("Please enter student`s English score: ")
    var english = scanner.nextInt()
    println("Please enter student`s Math score: ")
    var math = scanner.nextInt()
    val student = Student(name, english, math)
    student.print()
    student.nameCheck()
}

class GraduateStudent(name: String?, english: Int, math: Int, var thesis : Int) : Student(name, english, math){
    companion object{
        var pass = 70
    }

    override fun print() {
        println("$name\t$english\t$math\t$thesis\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }

    override fun passOrFailed() = if (getAverage() >= pass) "PASS" else "FAILED"
}

open class Student(var name: String?, var english: Int, var math: Int) {
    companion object{
        @JvmStatic
        var pass = 60
        fun test(){
            println("testing")
        }
    }

    open fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }

    fun grading() = when(getAverage()){
        in 90..100 -> 'A'
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else -> 'F'
    }

    open fun passOrFailed() = if (getAverage() >= pass) "PASS" else "FAILED"

    fun getAverage() = (english + math) / 2

    fun highest() = if (english > math) {
        println("English")
        english
    } else{
        println("Math")
        math
    }

    fun nameCheck() =  println(name?.length)
}
